src/main/java/com/zufar/icedlatte/payment/converter/PaymentConverter.java

```diff
@@ -2,19 +2,69 @@
 
 import com.zufar.icedlatte.cart.converter.ShoppingCartItemDtoConverter;
 import com.zufar.icedlatte.cart.entity.ShoppingCartItem;
+import com.zufar.icedlatte.openapi.dto.PaymentStatus;
 import com.zufar.icedlatte.openapi.dto.ProcessedPaymentDetailsDto;
+import com.zufar.icedlatte.openapi.dto.ShoppingCartItemDto;
 import com.zufar.icedlatte.payment.entity.Payment;
-import org.mapstruct.Mapper;
-import org.mapstruct.Mapping;
-import org.mapstruct.MappingConstants;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
 
+import java.util.LinkedHashSet;
 import java.util.Set;
 
-@Mapper(uses = ShoppingCartItemDtoConverter.class , componentModel = MappingConstants.ComponentModel.SPRING)
-public interface PaymentConverter {
+@Service
+@RequiredArgsConstructor
+public class PaymentConverter {
 
-    @Mapping(target = "items", source = "shoppingCartItems", qualifiedByName = {"toShoppingCartItemDto"})
-    ProcessedPaymentDetailsDto toDto(final Payment payment,
-                                     final Set<ShoppingCartItem> shoppingCartItems);
+    private final ShoppingCartItemDtoConverter shoppingCartItemDtoConverter;
 
-}
+    public ProcessedPaymentDetailsDto toDto(final Payment payment,
+                                     final Set<ShoppingCartItem> shoppingCartItems) {
+        if ( payment == null && shoppingCartItems == null ) {
+            return null;
+        }
+
+        ProcessedPaymentDetailsDto processedPaymentDetailsDto = new ProcessedPaymentDetailsDto();
+
+        if ( payment != null ) {
+            processedPaymentDetailsDto.setPaymentId( payment.getPaymentId() );
+            processedPaymentDetailsDto.setItemsTotalPrice( payment.getItemsTotalPrice() );
+            processedPaymentDetailsDto.setPaymentIntentId( payment.getPaymentIntentId() );
+            processedPaymentDetailsDto.setStatus( paymentStatusToPaymentStatus( payment.getStatus() ) );
+            processedPaymentDetailsDto.setDescription( payment.getDescription() );
+        }
+        processedPaymentDetailsDto.setItems( shoppingCartItemSetToShoppingCartItemDtoSet( shoppingCartItems ) );
+
+        return processedPaymentDetailsDto;
+    }
+
+    protected Set<ShoppingCartItemDto> shoppingCartItemSetToShoppingCartItemDtoSet(Set<ShoppingCartItem> set) {
+        if ( set == null ) {
+            return null;
+        }
+
+        Set<ShoppingCartItemDto> set1 = new LinkedHashSet<ShoppingCartItemDto>( Math.max( (int) ( set.size() / .75f ) + 1, 16 ) );
```

Comment:
nit: what are these magic numbers? Could you extract them to constants?

Reply:
if be honest, i dnk, i just made a copy of mapStruct's class)